source/vir/src/well_formed.rs

```diff
@@ -1412,7 +1457,7 @@ pub fn check_crate(
 
     let ctxt = Ctxt { funs, reveal_groups, dts, krate: krate.clone(), unpruned_krate };
     for function in krate.functions.iter() {
-        check_function(&ctxt, function, diags, no_verify)?;
+        check_function(&ctxt, function, diags, no_verify, no_cheating)?;
```

Comment:
I think you can stash `no_cheating` in the `well_formed` `Ctxt` (the first argument here) so you don't have to pass it as a separate argument to all the `check_` functions.